modules/cpr/src/main/java/org/atmosphere/container/AbstractJetty9AsyncSupportWithWebSocket.java

```diff
@@ -0,0 +1,173 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.lang.reflect.Method;
+
+abstract class AbstractJetty9AsyncSupportWithWebSocket extends Servlet30CometSupport {
+
+    private static final Logger staticLogger = LoggerFactory.getLogger(AbstractJetty9AsyncSupportWithWebSocket.class);
+
+    private final Logger logger;
+
+
+    static WebSocketPolicy GetPolicy(final AtmosphereConfig config) {
+        String bs = config.getInitParameter(ApplicationConfig.WEBSOCKET_BUFFER_SIZE);
+        WebSocketPolicy policy = new WebSocketPolicy(WebSocketBehavior.SERVER);
+        if (bs != null) {
+            policy.setInputBufferSize(Integer.parseInt(bs));
+        }
+
+        String max = config.getInitParameter(ApplicationConfig.WEBSOCKET_IDLETIME);
+        if (max != null) {
+            policy.setIdleTimeout(Integer.parseInt(max));
+        }
+
+        try {
+            // Crazy Jetty API Incompatibility
+            String serverInfo = config.getServletConfig().getServletContext().getServerInfo();
+            boolean isJetty91Plus = false;
+            if (serverInfo != null) {
+                int version = Integer.valueOf(serverInfo.split("/")[1].substring(0, 3).replace(".", ""));
+                isJetty91Plus = version > 90;
+            }
+
+            max = config.getInitParameter(ApplicationConfig.WEBSOCKET_MAXTEXTSIZE);
+            if (max != null) {
+                //policy.setMaxMessageSize(Integer.parseInt(max));
+                Method m;
+                if (isJetty91Plus) {
+                    m = policy.getClass().getMethod("setMaxTextMessageSize", new Class[]{int.class});
+                } else {
+                    m = policy.getClass().getMethod("setMaxMessageSize", new Class[]{long.class});
+                }
+                m.invoke(policy, Integer.parseInt(max));
+            }
+
+            max = config.getInitParameter(ApplicationConfig.WEBSOCKET_MAXBINARYSIZE);
+            if (max != null) {
+                //policy.setMaxMessageSize(Integer.parseInt(max));
+                Method m;
+                if (isJetty91Plus) {
+                    m = policy.getClass().getMethod("setMaxBinaryMessageSize", new Class[]{int.class});
+                } else {
+                    m = policy.getClass().getMethod("setMaxMessageSize", new Class[]{long.class});
+                }
+                m.invoke(policy, Integer.parseInt(max));
+            }
+        } catch (Exception ex) {
+            staticLogger.warn("", ex);
+        }
+
+        return policy;
+    }
+
+    AbstractJetty9AsyncSupportWithWebSocket(AtmosphereConfig config, Logger logger) {
+        super(config);
+        this.logger = logger;
+    }
+
+    WebSocketCreator buildCreator(final HttpServletRequest request, final HttpServletResponse response, final WebSocketProcessor webSocketProcessor) {
+        return new WebSocketCreator() {
+
+            // @Override  9.0.x
```

Comment:
Due to the evolution of permessage-deflate extension support in Chrome and Firefox, all versions of Jetty websocket prior to Jetty 9.3.8 are now no-longer supported on Chrome and Firefox.   Consider making Jetty 9.3.8 your minimum support level as well (or disable permessage-deflate support in Jetty versions prior to Jetty 9.3.8)